include/lsst/afw/math/PixelScaleBoundedField.h

```diff
@@ -0,0 +1,103 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/** \file
+ * A BoundedField that gives the amount a pixel is distorted at each point.
+ */
+#ifndef LSST_AFW_MATH_PixelScaleBoundedField_h_INCLUDED
+#define LSST_AFW_MATH_PixelScaleBoundedField_h_INCLUDED
+
+#include "lsst/afw/geom/SkyWcs.h"
+#include "lsst/afw/math/BoundedField.h"
+#include "lsst/pex/exceptions.h"
+
+namespace lsst {
+namespace afw {
+namespace math {
+
+/**
+ * A BoundedField that gives the amount a pixel is distorted at each point.
+ *
+ * This is ratio of the SkyWcs-transformed pixel area to the transformed pixel area at the SkyWcs center, or
+ * equivalently the determinant of the Jacobian of the SkyWcs Transform.
+ *
+ * Typically used to move an image or source flux between surface brightness and fluence space.
+ */
+class PixelScaleBoundedField : public table::io::PersistableFacade<PixelScaleBoundedField>,
+                               public BoundedField {
+public:
+    /**
+     *  Create a PixelScaleBoundedField from a bounding box and SkyWcs.
+     */
+    PixelScaleBoundedField(lsst::geom::Box2I const &bbox, geom::SkyWcs const &skyWcs)
+            : BoundedField(bbox),
+              _skyWcs(skyWcs),
+              _inverseScale(1.0 / std::pow(skyWcs.getPixelScale().asDegrees(), 2)) {}
+
+    ~PixelScaleBoundedField() override = default;
+
+    PixelScaleBoundedField(PixelScaleBoundedField const &) = default;
+    PixelScaleBoundedField(PixelScaleBoundedField &&) = default;
+    PixelScaleBoundedField &operator=(PixelScaleBoundedField const &) = delete;
+    PixelScaleBoundedField &operator=(PixelScaleBoundedField &&) = delete;
+
+    /// Get the contained SkyWcs
+    geom::SkyWcs const &getSkyWcs() const { return _skyWcs; }
+    // Get the cached inverse pixel scale
+    double getDefaultInverseScale() const { return _inverseScale; }
+
+    /// @copydoc BoundedField::evaluate
+    double evaluate(lsst::geom::Point2D const &position) const override;
+
+    /// TransformBoundedField is not persistable.
+    bool isPersistable() const noexcept override { return false; }
+
+    /// @copydoc BoundedField::operator*
+    std::shared_ptr<BoundedField> operator*(double const scale) const override {
+        throw LSST_EXCEPT(pex::exceptions::LogicError, "Not implemented");
+    }
+
+    /// @copydoc BoundedField::operator==
+    bool operator==(BoundedField const &rhs) const override;
+
+    // protected:
+    // std::string getPersistenceName() const override;
+
+    // std::string getPythonModule() const override;
+
+    // void write(OutputArchiveHandle &handle) const override;
```

Comment:
How is the persistence done if these are commented out?
And the persistence isn't tested.

Reply:
PixelScaleBoundedField is not persistable: `isPersistable()` is `false` See comments on the ticket: it will likely only be used in-memory, and is trivially constructable from a SkyWcs.

Reply:
So is it necessary to inherit from `table::io::PersistableFacade`? If so, there should be a comment explaining why.